Deeploy/Targets/Generic/Layers.py

```diff
@@ -618,3 +618,67 @@ class DequantLayer(ONNXLayer):
 
     def __init__(self, maps: List[NodeMapper]):
         super().__init__(maps)
+
+
+class BatchNormalizationLayer(ONNXLayer):
+
+    def __init__(self, maps: List[NodeMapper]):
+        super().__init__(maps)
+
+    def computeOps(self):
+        # 5 operations per element: sub, mul, add, sqrt, div
+        B = self.mapper.parser.operatorRepresentation['batch_size']
+        C = self.mapper.parser.operatorRepresentation['channel_size']
+        W = self.mapper.parser.operatorRepresentation['window_size']
+        return B * C * W * 5
+
+
+class ConvTransposeLayer(ONNXLayer):
+
+    def __init__(self, maps: List[NodeMapper]):
+        super().__init__(maps)
+
+    def computeShapes(self, inputShapes: Shape, outputShapes: Shape, operatorRepresentation,
+                      channels_first) -> Tuple[Shape, Shape]:
+        """
+        Infers output shapes for ConvTranspose using only static info.
+        - inputShapes[0]: input tensor shape (e.g., [N, C_in, W] for 1D, [N, C_in, H, W] for 2D)
+        - inputShapes[1]: weight tensor shape (e.g., [C_in, C_out // group, kW] for 1D)
+        - outputShapes[0]: output tensor shape (to be updated)
+        """
+        newInputShapes = list(inputShapes)
+        newOutputShapes = list(outputShapes)
+        group = operatorRepresentation.get('group', 1)
+        weight_shape = inputShapes[1]
+
+        if newOutputShapes and len(newOutputShapes[0]) >= 2:
+            if channels_first:
+                # For 1D: weight_shape = [C_in, C_out // group, kW]
+                # For 2D: weight_shape = [C_in, C_out // group, kH, kW]
+                ch_out = weight_shape[1] * group
+                newOutputShapes[0][1] = ch_out
+            else:
+                # For 1D: weight_shape = [C_in, C_out // group, kW]
+                # For 2D: weight_shape = [C_in, C_out // group, kH, kW]
+                ch_out = weight_shape[-2] * group
```

Comment:
This is not consistent with the comment above. It should also be `ch_out = weight_shape[1] * group`.

Reply:
Fixed in 2709d185a2aeb2739ef0f50bf94487596cfca81d